packages/firestore/src/api/database.ts

```diff
@@ -2652,6 +2682,16 @@ function resultChangeType(type: ChangeType): firestore.DocumentChangeType {
   }
 }
 
+function isEqualConverter(
+  first: firestore.FirestoreDataConverter<unknown> | undefined,
+  second: firestore.FirestoreDataConverter<unknown> | undefined
+): boolean {
+  if (first === undefined) {
+    return second === undefined;
+  } else {
+    return second !== undefined && first.toString() === second.toString();
```

Comment:
I originally tried comparing with `first === second`, but it caused the equality tests to fail because I think it compares the memory at which the function is defined rather than the internals of the function. I read about other ways to make equality work by putting it on the Object prototype, but the `toString()` way seemed the most straightforward.

Reply:
Hrm!  So the "compares the memory at which the function is defined rather than the internals of the function" is what I was expecting.  So if you have two converters that happen to be identical but are defined separately in your code, then they wouldn't compare equal.  I think this would be OK.

That said, comparing .toString() is perhaps better.  I was initially skeptical because I thought it was essentially undefined behavior (i.e. browser-dependent), but I guess it was standardized (maybe in ES2015? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/toString#Specifications) and so I guess it's okay to rely on it.

I'm still a tiny bit hesitant, but I don't have a good reason.  My biggest argument would perhaps be that it might be kinda' slow (I'm not sure how the browser generates the string). But it's probably fine.

Reply:
API discussion decision was to use `first === second` direct comparison.